internal/metrics/metrics.go

```diff
@@ -67,7 +67,18 @@ var (
 			Name:      "upstream_rpc_request_errors_total",
 			Help:      "Count of total errors when forwarding RPC requests to upstreams.",
 		},
-		[]string{"endpoint_id", "url", "response_code", "jsonrpc_error_code"},
+		[]string{"client", "endpoint_id", "url", "jsonrpc_method", "response_code", "jsonrpc_error_code"},
+	)
+
+	UpstreamRPCDuration = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: metricsNamespace,
+			Subsystem: metricsSubsystem,
+			Name:      "upstream_rpc_duration_seconds",
+			Help:      "Latency of RPC requests forwarded to upstreams.",
+			Buckets:   []float64{0.1, .5, 1, 5, 10, 30, 60},
```

Comment:
I suspect the histogram buckets are not granular enough b/c most will be under 1 second.

I'm about to put up a PR, but i'm using `Buckets:   []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10, 20, 40},`